File: primerTrimestre/introduccion_programacion_python/3repetitivas_py/ej12rep.py
# Programa: ej12rep.py
# Propósito: Pide una cadena y dos caracteres por teclado (valida que sea un carácter), 
# sustituye la aparición del primer carácter en la cadena por el segundo carácter.
#
# Autor: Antonio Quesada
# Fecha: 30/10/2019.
#
# Variables a usar:
#   * cadena
#   * caracter1
#   * caracter2

#Pedimos que introduzca una cadena.
cadena = str(input("Introduce una cadena:"))
caracter1 = str(input("Introduce el primer carácter de la cadena:"))
caracter2 = str(input("Introduce otro carácter:"))

for i in range(0,1):
    if len(caracter1) == 1 and len(caracter2) == 1:
        print (cadena[0].replace(caracter1,caracter2)+cadena)

File: primerTrimestre/java basico/3repetitivas/Ej12rep.java
/**Pide una cadena y dos caracteres por teclado (valida que sea un carácter), sustituye la aparición del primer
 * carácter en la cadena por el segundo carácter.
 *
 * @author Antonio Quesada
 */

import java.util.Scanner;

public class Ej12rep {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    
    //Leemos una cadena.
    System.out.println("Introduce una cadena: ");
    String cadena = s.nextLine();
    
    //Leemos el caracter de la cadena que queremos sustituir.
    System.out.println("Introduce el caracter a sustituir: ");
    String caracter1 = s.nextLine();
    
    //Leemos el caracter nuevo que vamos a sustituir en la cadena.
    System.out.println("Introduce el caracter nuevo a sustituir en la cadena: ");
    String caracter2 = s.nextLine();
    
    //Bucle do-while.
    do { //Reemplazamos el caracter 1 or el 2 mientras que la longitud de la cadena sea 1.
      String cadenaModificada = cadena.replace(caracter1, caracter2);   
      System.out.println("Se ha cambiado en la cadena "+caracter1+" por "+caracter2);
      System.out.println("La cadena final es "+cadenaModificada);
    } while(caracter1.length()!=1 && caracter2.length()!=1);
      
    } 
      
    }
    
  
